ml-agents/mlagents/trainers/tf_policy.py

```diff
@@ -169,6 +177,24 @@ def make_empty_memory(self, num_agents):
         """
         return np.zeros((num_agents, self.m_size))
 
+    def save_memories(self, agent_ids, memory_matrix):
+        if not isinstance(memory_matrix, np.ndarray):
+            return
+        for index, id in enumerate(agent_ids):
+            self.memory_dict[id] = memory_matrix[index, :]
+
+    def retrieve_memories(self, agent_ids):
+        memory_matrix = np.zeros((len(agent_ids), self.m_size))
+        for index, id in enumerate(agent_ids):
+            if id in self.memory_dict.keys():
```

Comment:
This will iterate through the keys and check for id. 
```suggestion
            if id in self.memory_dict:
```

Reply:
I was wrong about the performance.